system-addon/content-src/components/Base/Base.jsx

```diff
@@ -22,6 +23,9 @@ function addLocaleDataForReactIntl({locale}) {
 
 class Base extends React.Component {
   componentDidMount() {
+    // Request state after the first render
+    this.props.dispatch(ac.SendToMain({type: at.NEW_TAB_STATE_REQUEST}));
+
```

Comment:
A comment explaining why we do this extra round trip (e.g. for pre-rendering), perhaps even in a bit of details, would likely be useful to future code readers.

Reply:
A unit test for this would be good too; https://github.com/airbnb/enzyme/blob/master/docs/api/mount.md has an example using sinon that should be similar.